plugin/trino-hive/src/test/java/io/trino/plugin/hive/parquet/write/TestMapredParquetOutputFormat.java

```diff
@@ -40,6 +43,17 @@
 public class TestMapredParquetOutputFormat
         extends MapredParquetOutputFormat
 {
+    static {
+        try {
```

Comment:
why this test & this test only?

Reply:
This change is needed only because the tests using this don't go via `ParquetRecordWriter` for writing parquet files with old writer. This class is used by some tests for greater control over the encodings used in parquet schema.

Reply:
what about putting the static code in some other, shared class `ReplaceHadoopParquetMemoryManager`

and then call `ReplaceHadoopParquetMemoryManager.ensureReplaced`

maybe then you even don't need to do this in a static initializer at all?
just have a static (synchronized) boolean field saying whether you did that already?

Reply:
I've re-used the code now through a static method.
I'm doing this in a static initialiser because we need the `memoryManager` reference replaced with our implementation before any writers are created, so that the Hadoop code below from `org.apache.parquet.hadoop.ParquetOutputFormat#getRecordWriter` won't try to make it's own memory manager
```
synchronized (ParquetOutputFormat.class) {
      if (memoryManager == null) {
        memoryManager = new MemoryManager(maxLoad, minAllocation);
      }
}
```